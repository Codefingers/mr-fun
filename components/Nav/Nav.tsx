import React from 'react';
import styled from 'styled-components';

const NavContainer = styled.ul`
  background: #1D2433;
  list-style: none;
  padding: 0 32px;
  margin: 0;

  li {
    display: inline-block;
  }

  a {
    display: block;
    padding: 16px;
    color: #838FA6;
    text-decoration: none;

    &:hover {
      color: #C5CCD9;
    }
  }
`;

export const Nav = () => (
  <NavContainer>
    <li>
      <a href="/review">Part 1: Review</a>
    </li>
    <li>
      <a href="/task">Part 2: Task</a>
    </li>
  </NavContainer>
);
