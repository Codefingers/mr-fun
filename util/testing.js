import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

const Providers = ({ children }) => {
  return (
    <ThemeProvider
      theme={{
        colors: {
          body: '#838fa6',
          heading: '#ebedf2',
          accent: '#BA99FF',
          button: '#FFF',
        },
      }}
    >
      {children}
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

// override render method
export { customRender as render };
