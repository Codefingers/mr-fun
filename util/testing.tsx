import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../theme';

export const Providers = ({ children }) => {
  return (
    <ThemeProvider
      theme={theme}
    >
      {children}
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

// override render method
export { customRender as render };
