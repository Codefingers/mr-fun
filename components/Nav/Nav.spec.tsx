import React from 'react';
import { render } from '../../util/testing';

import { Nav } from './Nav';

describe('<Nav />', () => {
  it('renders', () => {
    const { container } = render(<Nav />);
    expect(container).toMatchSnapshot();
  });
});
