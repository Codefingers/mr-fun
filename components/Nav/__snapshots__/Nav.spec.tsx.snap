// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders 1`] = `
<div>
  <ul
    class="Nav__NavContainer-sc-1gxm0jw-0 hppfIP"
  >
    <li>
      <a
        href="/review"
      >
        Part 1: Review
      </a>
    </li>
    <li>
      <a
        href="/task"
      >
        Part 2: Task
      </a>
    </li>
  </ul>
</div>
`;
